generate-geodatabase-replica-from-feature-service/src/main/java/com/esri/arcgismaps/sample/generategeodatabasereplicafromfeatureservice/MainActivity.kt

```diff
@@ -0,0 +1,259 @@
+        // representing portland area
+        map.maxExtent = Envelope(
+            -13687689.2185849,
+            5687273.88331375,
+            -13622795.3756647,
+            5727520.22085841,
+            spatialReference = SpatialReference.webMercator()
+        )
+        // configure mapview assignments
+        mapView.apply {
+            this.map = map
+            // add the graphics overlay to display the boundary
+            graphicsOverlays.add(graphicsOverlay)
+        }
+
+        // create a geodatabase sync task with the feature service url
+        // This feature service shows a web map of portland street trees,
+        // their attributes, as well as related inspection information
+        val geodatabaseSyncTask = GeodatabaseSyncTask(getString(R.string.feature_server_url))
+
+        // set the button's onClickListener
+        generateButton.setOnClickListener {
+            mapView.visibleArea?.let { polygon ->
+                // start the geodatabase generation process
+                generateGeodatabase(geodatabaseSyncTask, map, polygon.extent)
+            }
+        }
+
+        lifecycleScope.launch {
+            // show the error and return if map load failed
+            map.load().onFailure {
+                showError("Unable to load map")
+                return@launch
+            }
+
+            geodatabaseSyncTask.load().onFailure {
+                // if the metadata load fails, show the error and return
+                showError("Failed to fetch geodatabase metadata")
+                return@launch
+            }
+
+            // enable the generate button since the task is now loaded
+            generateButton.isEnabled = true
+        }
+    }
+
+    /**
+     * Starts a [geodatabaseSyncTask] with the given [map] and [extents],
+     * runs a GenerateGeodatabaseJob and saves the geodatabase file into local storage
+     */
+    private fun generateGeodatabase(
+        geodatabaseSyncTask: GeodatabaseSyncTask,
+        map: ArcGISMap,
+        extents: Envelope
+    ) {
+        // clear any layers already on the map
+        map.operationalLayers.clear()
+        // clear all symbols drawn
+        graphicsOverlay.graphics.clear()
+
+        // create a boundary representing the extents selected
+        val boundary = Graphic(extents, boundarySymbol)
+        // add this boundary to the graphics overlay
+        graphicsOverlay.graphics.add(boundary)
+
+        lifecycleScope.launch {
+            // create generate geodatabase parameters for the selected extents
+            val defaultParameters =
+                geodatabaseSyncTask.createDefaultGenerateGeodatabaseParameters(extents).getOrElse {
+                    // show the error and return if the task fails
+                    showError("Error creating geodatabase parameters")
+                    return@launch
+                }
+
+            // set return attachments option to false
```

Comment:
It might be worth adding what this means, for users who are not familiar why this should be false.